docs/modeling/example-fitting-constraints.rst

```diff
@@ -64,11 +64,14 @@ Bounded Constraints
 
 Bounded fitting is supported through the ``bounds`` arguments to models or by
 setting `~astropy.modeling.Parameter.min` and `~astropy.modeling.Parameter.max`
-attributes on a parameter.  Bounds for the
-`~astropy.modeling.fitting.LevMarLSQFitter` are always exactly satisfied--if
-the value of the parameter is outside the fitting interval, it will be reset to
-the value at the bounds. The `~astropy.modeling.fitting.SLSQPLSQFitter` optimization
-algorithm handles bounds internally.
+attributes on a parameter. The following fitters support bounds internally:
+
+* `~astropy.modeling.fitting.TRFLSQFitter`
+* `~astropy.modeling.fitting.DogBoxLSQFitter`
+* `~astropy.modeling.fitting.SLSQPLSQFitter`
+
+The `~astropy.modeling.fitting.LevMarLSQFitter` algorithm uses an unsophisticated
+method of handling bounds and is no longer recommended.
```

Comment:
[citation needed]

Where does this argument come from?  Has it been demonstrated?  I think yes, but we need to link to source material that shows it.  Telling users not to use the most popular (historically) method, but not telling them why or under what circumstances to adopt each of the others, is unhelpful.

I am quite interested in the differences between these, but I am totally overwhelmed right now - if you catch me at a better time, I'll read literature & write a summary.  But if you can provide more information now, do it!

Reply:
I'll try and write up something more soon but I should probably rephrase this - the LM algorithm doesn't actually implement support for bounds, so we just hackily reset any out of bounds parameters to the min or max during each iteration in our fitter class, which is why it is unsophisticated.

Reply:
Here is a citation, from existing docs :sweat_smile: 

https://docs.astropy.org/en/stable/modeling/fitting.html#notes-on-non-linear-fitting

*This fitter supports parameter bounds via an unsophisticated min/max condition which can cause parameters to “stick” to one of the bounds if during the fitting process the parameter gets close to the bound during some of the intermediate fitting operations.*

Basically you won't find papers as such discussing this, we just implement a crude approach to bounds in our own fitting:

https://github.com/astropy/astropy/blob/d6020404d2b9a98300cbf3070386857416fb7968/astropy/modeling/fitting.py#L2049-L2054

Basically during the iteration, if a parameter goes too high or too low we just reset it to the min/max bound. This can cause issues in some use cases as the parameters can just end up getting stuck against the bound instead of exploring the rest of parameter space.

In contrast, algorithms such as the 'Trust Region Reflective' algorithm have more carefully thought out ways of dealing with bounds:

https://link.springer.com/article/10.1007/BF01582221

I will try and edit the docs to be a bit more specific than just saying sophisticated/unsophisticated.

Reply:
Also this: https://github.com/astropy/astropy/issues/11946

Reply:
Thanks.  It wasn't clear to me before what the astropy approach was.  This is helpful!